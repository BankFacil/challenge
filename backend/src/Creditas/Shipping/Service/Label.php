<?php

namespace Creditas\Shipping\Service;

use Creditas\Shipping\Enum\Messages;

class Label
{
    public function generate(Payment $payment, $noTax = false)
    {
        $message = Messages::SHIPPING_LABEL_DEFAULT;

        if ($noTax) {
            $message = Messages::SHIPPING_LABEL_NO_TAX;
        }

        // @TODO: generate shipping label with message
    }    
}
