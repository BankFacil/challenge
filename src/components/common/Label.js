import styled from 'styled-components/native';

import theme from '../theme';

const Label = styled.Text`
  text-align: left;
  color: ${p => theme.getColorFromProps(p)};
  margin: ${theme.spacing.margin / 2}px;
`;

export default Label;
