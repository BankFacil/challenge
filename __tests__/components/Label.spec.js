import React from 'react';
import renderer from 'react-test-renderer';

import Label from '../../src/components/common/Label';

describe('Label', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const tree = renderer.create(<Label>Testing</Label>).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should match to snapshot - light color', () => {
      const tree = renderer.create(<Label light>Testing</Label>).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
