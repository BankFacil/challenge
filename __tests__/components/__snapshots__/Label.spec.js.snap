// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Rendering should match to snapshot - light color 1`] = `
<Text
  light={true}
  style={
    Array [
      Object {
        "color": "#fff",
        "marginBottom": 6,
        "marginLeft": 6,
        "marginRight": 6,
        "marginTop": 6,
        "textAlign": "left",
      },
    ]
  }
>
  Testing
</Text>
`;

exports[`Label Rendering should match to snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#3099DD",
        "marginBottom": 6,
        "marginLeft": 6,
        "marginRight": 6,
        "marginTop": 6,
        "textAlign": "left",
      },
    ]
  }
>
  Testing
</Text>
`;
